beniget/beniget.py

```diff
@@ -463,26 +531,50 @@ def SwitchScopeContext(self, defs, scopes, scope_depths, precomputed_locals):
         self._scopes = scopes
         self._scope_depths = scope_depths
         self._precomputed_locals = precomputed_locals
-
+    
+    def process_functions_bodies(self):
+        for fnode, defs, scopes, scope_depths, precomputed_locals in self._defered:
+            visitor = getattr(self,
+                              "visit_{}".format(type(fnode).__name__))
+            with self.SwitchScopeContext(defs, scopes, scope_depths, precomputed_locals):
+                visitor(fnode, step=DefinitionStep)
+
+    def process_annotations(self):
+        compute_defs, self.defs = self.defs,  self.compute_annotation_defs
+        for annnode, heads, cb in self._defered_annotations[-1]:
+            visitor = getattr(self,
+                                "visit_{}".format(type(annnode).__name__))
+            currenthead, self._scopes = self._scopes, heads
+            cb(visitor(annnode)) if cb else visitor(annnode)
+            self._scopes = currenthead
+        self.defs = compute_defs
 
     # stmt
     def visit_Module(self, node):
         self.module = node
+
+        futures = collect_future_imports(node)
+        # determine whether the PEP563 is enabled
+        # allow manual enabling of DefUseChains.future_annotations
+        self.future_annotations |= 'annotations' in futures
+        
+
         with self.ScopeContext(node):
 
+
             self._definitions[-1].update(
                 {k: ordered_set((v,)) for k, v in self._builtins.items()}
             )
 
+            self._defered_annotations.append([])
             self.process_body(node.body)
 
             # handle function bodies
-            for fnode, defs, scopes, scope_depths, precomputed_locals in self._defered:
-                visitor = getattr(self,
-                                  "visit_{}".format(type(fnode).__name__))
-                with self.SwitchScopeContext(defs, scopes, scope_depths,
-                                             precomputed_locals):
-                    visitor(fnode, step=DefinitionStep)
+            self.process_functions_bodies()
+
+            # handle defered annotations as in from __future__ import annotations
+            self.process_annotations()
```

Comment:
Do you really want to process annotations at the end of each scope? From PEP563

> Note that as per [PEP 526](https://peps.python.org/pep-0526), local variable annotations are not evaluated at all since they are not accessible outside of the function’s closure.

Which may be understand as "we only care about values at the module level".